nodejs/eks/nodegroup.ts

```diff
@@ -1742,6 +1742,9 @@ function createManagedNodeGroupInternal(
     let launchTemplate: aws.ec2.LaunchTemplate | undefined;
     if (args.kubeletExtraArgs || args.bootstrapExtraArgs || args.enableIMDSv2) {
         launchTemplate = createMNGCustomLaunchTemplate(name, args, core, parent, provider);
+
+        // Disk size is specified in the launch template.
+        delete nodeGroupArgs.diskSize;
```

Comment:
Not quite following control flow here, is diskSize deleted here coming from what the user supplied? Should we warn the user that the supplied value is a no-op in this case?